src/librustc/session/mod.rs

```diff
@@ -1103,11 +1099,12 @@ pub fn build_session_with_source_map(
         },
     );
 
-    build_session_(sopts, local_crate_source_file, diagnostic_handler, source_map)
+    build_session_(sopts, self_profiler, local_crate_source_file, diagnostic_handler, source_map)
 }
 
 pub fn build_session_(
     sopts: config::Options,
+    self_profiler: Option<Arc<Mutex<SelfProfiler>>>,
```

Comment:
Why is `self_profiler` passed an argument here?

Reply:
We need it to pass on line 1189: https://github.com/rust-lang/rust/pull/58488/files#diff-1c226b200b9385d77428f0b4418366e4R1189

Reply:
Why isn't it created there like it was before?

Reply:
We wanted to create it as early as possible so we didn't lose time at the very beginning of the process. So we're creating it here https://github.com/rust-lang/rust/pull/58488/commits/7cc697641d7050682f9f1c768529d3839276269a#diff-707a0eda6b2f1a0537abc3d23133748cR466

Reply:
Yes, what was the number you had, @wesleywiser? Something like 30% of the time for compiling small crates was unaccounted for...

Reply:
I think it was closer to 10% for `hello_world.rs`. In practice, for a reasonably sized crate like `regex` or something, it was < 2%.